src/libs/workspace-utils.ts

```diff
@@ -180,10 +180,25 @@ export const isValidWsExportTarget = safeCurry((sourceWs: WorkspaceWrapper, dest
   );
 });
 
-// Returns a message explaining why the user can't edit the workspace, or undefined if they can
-export const editWorkspaceError = ({ accessLevel, workspace: { isLocked } }) => {
-  return cond(
-    [!canWrite(accessLevel), () => 'You do not have permission to modify this workspace.'],
-    [isLocked, () => 'This workspace is locked.']
+export interface WorkspaceAccessInfo {
+  accessLevel: WorkspaceAccessLevel;
+  workspace: { isLocked: boolean };
+}
+
+export const canEditWorkspace = ({
```

Comment:
lets get some unit tests in for these into workspace-utils.test.ts

Reply:
Can you make a suggestion? Since this function is a thin wrapper around `cond`, I don't see how to write a test that adds net value.

Reply:
take a look a the various 'code branches' that can produce true or false-with-message, have a small test for each branch that asserts the expected outcome.

Reply:
I stand corrected. I judge these do add value. https://github.com/DataBiosphere/terra-ui/pull/4268/commits/7917821e9565e200933767a836c5d50e741a11d0